client/src/components/ModalEditCategory.tsx

```diff
@@ -2,48 +2,56 @@ import { Dispatch, SetStateAction, useEffect, useState } from 'react';
 import { Modal, Pressable, StyleSheet, Text, View } from 'react-native';
 import EmojiPicker from 'rn-emoji-keyboard';
 import { EmojiType } from 'rn-emoji-keyboard/lib/typescript/types';
-import { changeCategoryStyle } from '../api/categoriesApi';
-import { Category } from '../constants/defaultCategories';
+import { ICategory, IUser } from '../constants/interfaces';
 import { STYLES } from '../styles/styles';
 import ModalColorPicker from './ModalColorPicker';
+import { changeCategory } from '../helpers/api';
 
 type ModalEditCategoryProps = {
   modalEditVisible: boolean;
   setModalEditVisible: Dispatch<SetStateAction<boolean>>;
-  setCategories: Function;
-  categories: Category[];
-  currentCategory: Category;
+  category: ICategory;
+  userId: string;
+  setUser: React.Dispatch<React.SetStateAction<IUser>>;
 };
 
 export default function ModalEditCategory({
   modalEditVisible,
   setModalEditVisible,
-  setCategories,
-  categories,
-  currentCategory,
+  category,
+  userId,
+  setUser,
 }: ModalEditCategoryProps) {
   const [isOpen, setIsOpen] = useState(false);
-  const [chosenIcon, setChosenIcon] = useState(currentCategory.icon);
-  const [chosenColor, setChosenColor] = useState(currentCategory.color);
+  const [chosenIcon, setChosenIcon] = useState(category.icon);
+  const [chosenColor, setChosenColor] = useState(category.color);
   const [modalColorVisible, setModalColorVisible] = useState(false);
+  const [error, setError] = useState('');
 
   useEffect(() => {
-    setChosenIcon(currentCategory.icon);
-    setChosenColor(currentCategory.color);
-  }, [currentCategory]);
+    setChosenIcon(category.icon);
```

Comment:
I believe it might be better to avoid useEffect here, this logic could be placed inside the submit handler or useCallback